expression/builtin_string.go

```diff
@@ -2360,6 +2360,8 @@ func (c *findInSetFunctionClass) getFunction(ctx sessionctx.Context, args []Expr
 	bf.tp.Flen = 3
 	sig := &builtinFindInSetSig{bf}
 	sig.setPbCode(tipb.ScalarFuncSig_FindInSet)
+	_, col, _ := sig.CharsetAndCollation(ctx)
+	sig.setCollator(collate.GetCollator(col))
```

Comment:
How about calling `setCollator` in the `newBaseBuiltinFuncWithTp` function?
Then we don't need to call it in each `getFunction` functions.